physics/GWD/drag_suite.F90

```diff
@@ -1414,6 +1414,1223 @@ subroutine drag_suite_run(                                           &
+         do k = kts,kpblmax
+            if (k .le. kbl(i)) taup(i,k) = taub(i)
+         enddo
+!
+         do k = kpblmin, km-1                   ! vertical level k loop!
+            kp1 = k + 1
+!
+!   unstablelayer if ri < ric
+!   unstable layer if upper air vel comp along surf vel <=0 (crit lay)
+!   at (u-c)=0. crit layer exists and bit vector should be set (.le.)
+!
+            if (k .ge. kbl(i)) then
+               icrilv(i) = icrilv(i) .or. ( usqj(i,k) .lt. ric)            &
+                                     .or. (velco(i,k) .le. 0.0)
+               brvf(i)  = max(bnv2(i,k),bnv2min) ! brunt-vaisala frequency squared
+               brvf(i)  = sqrt(brvf(i))          ! brunt-vaisala frequency
+            endif
+!
+            if (k .ge. kbl(i) .and. (.not. ldrag(i)))   then
+               if (.not.icrilv(i) .and. taup(i,k) .gt. 0.0 ) then
+                  temv = 1.0 / velco(i,k)
+                  tem1 = coefm(i)/dxy(i)*(ro(i,kp1)+ro(i,k))*brvf(i)*   &
+                                                      velco(i,k)*0.5
+                  hd   = sqrt(taup(i,k) / tem1)
+                  fro  = brvf(i) * hd * temv
+!
+!  rim is the minimum-richardson number by shutts (1985)
+                  tem2   = sqrt(usqj(i,k))
+                  tem    = 1. + tem2 * fro
+                  rim    = usqj(i,k) * (1.-fro) / (tem * tem)
+!
+!  check stability to employ the 'saturation hypothesis'
+!  of lindzen (1981) except at tropospheric downstream regions
+!
+                  if (rim .le. ric) then  ! saturation hypothesis!
+                        temc = 2.0 + 1.0 / tem2
+                        hd   = velco(i,k) * (2.*sqrt(temc)-temc) / brvf(i)
+                        taup(i,kp1) = tem1 * hd * hd
+                  else                    ! no wavebreaking!
+                     taup(i,kp1) = taup(i,k)
+                  endif
+               endif
+            endif
+         enddo
+!
+         if(lcap.lt.km) then
+            do klcap = lcapp1,km
+               taup(i,klcap) = prsi(i,klcap) / prsi(i,lcap) * taup(i,lcap)
+            enddo
+         endif
+
+      endif  ! if ( ls_taper(i).GT.1.E-02 )
+
+   enddo  ! do i=its,im
+
+ENDIF  ! (do_gsl_drag_ls_bl).and.(gwd_opt_ls .EQ. 1)
+!===============================================================
+!COMPUTE BLOCKING COMPONENT                                     
+!===============================================================
+IF ( (do_gsl_drag_ls_bl) .and. (gwd_opt_bl .EQ. 1) ) THEN
+   do i = its,im
+     flag(i) = .true.
+   enddo
+
+   do i=its,im
+
+      if ( ls_taper(i).GT.1.E-02 ) then
+
+         if (.not.ldrag(i)) then
+!
+!------- determine the height of flow-blocking layer
+!
+            pe = 0.0
+            ke = 0.0
+            do k = km, kpblmin, -1
+               if(flag(i).and. k.le.kbmax(i)) then
+                          pe = pe + bnv2(i,k)*(zl(i,kbmax(i))-zl(i,k))*      &
+                            del(i,k)/g/ro(i,k)
```

Comment:
You already have `g_inv` but are not using it here. Minor speedup opportunity.

Reply:
changed